stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -72,6 +124,125 @@ public View getViewForHighlighting(Object element) {
     return (View)element;
   }
 
+  @Override
+  protected void onGetStyles(View element, Accumulator<StyleInfo> styles) {
+    for (Method method : viewGetters) {
+      boolean caught = false;
+      try {
+        ViewDebug.ExportedProperty annotation = getExportedAnnotation(method);
+        Object returnedValue = method.invoke(element);
+        String cssName = convertGetterNameToCssName(method.getName());
+        String cssValue = String.valueOf(returnedValue);
+        boolean isValueDefault = false;
+
+        // special case id to translate it into a human-readable string
+        if (method.getName().equals("getId")) {
+          cssValue = getIdAttribute(element);
```

Comment:
We should strive to adhere to Single Assignment Form, which can be thought of as being part of Functional Programming Style. This method is writing to the value of `cssValue` multiple times, and this can make things very confusing when it comes time to read or debug it later.

Basically, stick a `final` on the front of each variable and find ways to transform your code to work within that constraint. In this case you can probably defer calling `String.valueOf(returnedValue)` (which is currently assigned to cssValue when it's declared) until much later, once you've ruled out the other alternatives.

This type of transformation isn't always straightforward if you're not accustomed to it, so I'm happy to help out with it. I promise it's worth it!